addons/account/models/account_move.py

```diff
@@ -2985,6 +2979,13 @@ def _compute_tax_audit(self):
     # CONSTRAINT METHODS
     # -------------------------------------------------------------------------
 
+    @api.constrains('currency_id', 'account_id')
+    def _check_account_currency(self):
+        for line in self:
+            account_currency = line.account_id.currency_id
+            if account_currency and account_currency != line.company_currency_id and account_currency != line.currency_id:
+                raise ValidationError(_('The selected account of your Journal Entry forces to provide a secondary currency. You should remove the secondary currency on the account.'))
```

Comment:
The account selected on your journal entry .....

(also change in translation, of course :p )